packages/core/backend/sirius-components-core-graphql/src/main/java/org/eclipse/sirius/components/core/graphql/datafetchers/ViewerOmniboxCommandsDataFetcher.java

```diff
@@ -0,0 +1,98 @@
+ *******************************************************************************/
+
+package org.eclipse.sirius.components.core.graphql.datafetchers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.sirius.components.annotations.spring.graphql.QueryDataFetcher;
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventProcessorRegistry;
+import org.eclipse.sirius.components.collaborative.dto.GetOmniboxCommandsInput;
+import org.eclipse.sirius.components.collaborative.dto.GetOmniboxCommandsPayload;
+import org.eclipse.sirius.components.collaborative.dto.OmniboxCommand;
+import org.eclipse.sirius.components.collaborative.dto.OmniboxContextEntry;
+import org.eclipse.sirius.components.collaborative.dto.PageInfoWithCount;
+import org.eclipse.sirius.components.graphql.api.IDataFetcherWithFieldCoordinates;
+
+import graphql.relay.Connection;
+import graphql.relay.ConnectionCursor;
+import graphql.relay.DefaultConnection;
+import graphql.relay.DefaultConnectionCursor;
+import graphql.relay.DefaultEdge;
+import graphql.relay.DefaultPageInfo;
+import graphql.relay.Edge;
+import graphql.relay.Relay;
+import graphql.schema.DataFetchingEnvironment;
+import reactor.core.publisher.Mono;
+
+/**
+ * Data fetcher for the field Viewer#omniboxCommands.
+ *
+ * @author gcoutable
+ */
+@QueryDataFetcher(type = "Viewer", field = "omniboxCommands")
+public class ViewerOmniboxCommandsDataFetcher implements IDataFetcherWithFieldCoordinates<CompletableFuture<Connection<OmniboxCommand>>> {
+
+    private static final String CONTEXT_ENTRIES_ARGUMENT = "contextEntries";
+
+    private static final String QUERY_ARGUMENT = "query";
+
+    private final IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    private final ObjectMapper objectMapper;
+
+    public ViewerOmniboxCommandsDataFetcher(IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry, ObjectMapper objectMapper) {
+        this.editingContextEventProcessorRegistry = Objects.requireNonNull(editingContextEventProcessorRegistry);
+        this.objectMapper = Objects.requireNonNull(objectMapper);
+    }
+
+    @Override
+    public CompletableFuture<Connection<OmniboxCommand>> get(DataFetchingEnvironment environment) throws Exception {
+        Object argument = environment.getArgument(CONTEXT_ENTRIES_ARGUMENT);
+        List<OmniboxContextEntry> omniboxContextEntries = this.objectMapper.convertValue(argument, new TypeReference<List<OmniboxContextEntry>>() { });
+        String query = environment.getArgument(QUERY_ARGUMENT);
+        var input = new GetOmniboxCommandsInput(UUID.randomUUID(), query);
+
+        return this.editingContextEventProcessorRegistry.dispatchEvent(omniboxContextEntries.get(0).id().toString(), input)
```

Comment:
Check the size and presence of a context entry with the editing context type instead of `.get(0)`

Reply:
Done